packages/nimble-components/src/anchor-button/tests/anchor-button.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { DOM, html } from '@microsoft/fast-element';
+import { AnchorButton } from '..';
+import { fixture, Fixture } from '../../utilities/tests/fixture';
+
+async function setup(): Promise<Fixture<AnchorButton>> {
+    return fixture<AnchorButton>(
+        html`<nimble-anchor-button></nimble-anchor-button>`
+    );
+}
+
+describe('AnchorButton', () => {
+    let element: AnchorButton;
+    let connect: () => Promise<void>;
+    let disconnect: () => Promise<void>;
+
+    beforeEach(async () => {
+        ({ element, connect, disconnect } = await setup());
+    });
+
+    afterEach(async () => {
+        await disconnect();
+    });
+
+    it('can construct an element instance', () => {
+        expect(document.createElement('nimble-anchor-button')).toBeInstanceOf(
+            AnchorButton
+        );
+    });
+
+    it('should set the "control" class on the internal control', async () => {
+        await connect();
+        expect(element.control.classList.contains('control')).toBe(true);
+    });
+
+    it('should set the `part` attribute to "control" on the internal control', async () => {
+        await connect();
+        expect(element.control.part.contains('control')).toBe(true);
+    });
+
+    it('should clear `href` on the internal control when disabled', async () => {
+        await connect();
+        element.control.setAttribute('href', 'http://www.ni.com');
+
+        element.disabled = true;
+        await DOM.nextUpdate();
+
+        expect(element.control.getAttribute('href')).toBeNull();
+    });
+
+    const reflectAttributeTestData: { attrName: string }[] = [
+        { attrName: 'download' },
+        { attrName: 'hreflang' },
+        { attrName: 'ping' },
+        { attrName: 'referrerpolicy' },
+        { attrName: 'rel' },
+        { attrName: 'target' },
+        { attrName: 'type' },
+        { attrName: 'aria-atomic' },
+        { attrName: 'aria-busy' },
+        { attrName: 'aria-controls' },
+        { attrName: 'aria-current' },
+        { attrName: 'aria-describedby' },
+        { attrName: 'aria-details' },
+        { attrName: 'aria-disabled' },
+        { attrName: 'aria-errormessage' },
+        { attrName: 'aria-expanded' },
+        { attrName: 'aria-flowto' },
+        { attrName: 'aria-haspopup' },
+        { attrName: 'aria-hidden' },
+        { attrName: 'aria-invalid' },
+        { attrName: 'aria-keyshortcuts' },
+        { attrName: 'aria-label' },
+        { attrName: 'aria-labelledby' },
+        { attrName: 'aria-live' },
+        { attrName: 'aria-owns' },
+        { attrName: 'aria-relevant' },
+        { attrName: 'aria-roledescription' }
+    ];
+    reflectAttributeTestData.forEach(testData => {
```

Comment:
If switching to a theory pattern can you use the patterns used in our other nimble tests?
https://github.com/ni/nimble/blob/54f0492a29685b21c438ae69d6fc4eb291896646/packages/nimble-components/src/icon-base/tests/icons.spec.ts#L68

The benefit is they provide the ability to conditionally disable or focus specific tests and aligns with our our theory test patterns.

Reply:
I was following the example of the combobox tests, but I guess that wasn't using the right pattern. Fixed.